chef-utils/lib/chef-utils/dsl/virtualization.rb

```diff
@@ -0,0 +1,231 @@
+      def parallels?(node = __getnode)
+        node.dig("virtualization", "system") == "parallels" && node.dig("virtualization", "role") == "guest"
+      end
+
+      # Determine if the current node is a Parallels Desktop host.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #   true if the machine is currently running under Parallels Desktop, false
+      #   otherwise
+      #
+      def parallels_host?(node = __getnode)
+        node.dig("virtualization", "system") == "parallels" && node.dig("virtualization", "role") == "host"
+      end
+
+      # Determine if the current node is a VirtualBox guest.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def vbox?(node = __getnode)
+        node.dig("virtualization", "system") == "vbox" && node.dig("virtualization", "role") == "guest"
+      end
+
+      # Determine if the current node is a VirtualBox host.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def vbox_host?(node = __getnode)
+        node.dig("virtualization", "system") == "vbox" && node.dig("virtualization", "role") == "host"
+      end
+
+      alias_method :virtualbox?, :vbox?
+
+      # Determine if the current node is a VMWare guest.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def vmware?(node = __getnode)
+        node.dig("virtualization", "system") == "vmware" && node.dig("virtualization", "role") == "guest"
+      end
+
+      # Determine if the current node is VMware host.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def vmware_host?(node = __getnode)
+        node.dig("virtualization", "system") == "vmware" && node.dig("virtualization", "role") == "host"
+      end
+
+      # Determine if the current node is an openvz guest.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def openvz?(node = __getnode)
+        node.dig("virtualization", "system") == "openvz" && node.dig("virtualization", "role") == "guest"
+      end
+
+      # Determine if the current node is an openvz host.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def openvz_host?(node = __getnode)
+        node.dig("virtualization", "system") == "openvz" && node.dig("virtualization", "role") == "host"
+      end
+
+      # Determine if the current node is running under any virutalization environment
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def guest?(node = __getnode)
+        node.dig("virtualization", "role") == "guest"
+      end
+
+      alias_method :virtual?, :guest?
+
+      # Determine if the current node is running under any virutalization environment
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def virtual_host?(node = __getnode)
```

Comment:
hypervisor?

Reply:
lxc is container based though and doesn't have a hypervisor so that word would be wrong, i think.

Reply:
oh yeah, i see that could be read either way though so that is also terrible

Reply:
i hate words

Reply:
vmhost, hypervisor...

Reply:
vmm or vmms which probably nobody would guess

Reply:
I think hypervisor is less terrible than vm_host, but given a time machine container stuff would go in node['container'] not node['virtualization']

Reply:
yeah i polled #general on slack and wound up at `hypervisor?` as being least-bad.